Index: Makefile
===================================================================
--- Makefile	(revision 19814)
+++ Makefile	(working copy)
@@ -38,6 +38,7 @@
               sub_cc.c \
               subreader.c \
               vobsub.c \
+              ../nut/libnut/demuxer.c \
 
 ifeq ($(UNRARLIB),yes)
 SRCS_COMMON += unrarlib.c
Index: libmpdemux/Makefile
===================================================================
--- libmpdemux/Makefile	(revision 19814)
+++ libmpdemux/Makefile	(working copy)
@@ -37,6 +37,7 @@
         demux_mpc.c \
         demux_mpg.c \
         demux_nsv.c \
+        demux_nut.c \
         demux_nuv.c \
         demux_pva.c \
         demux_rawaudio.c \
@@ -103,7 +104,7 @@
 
 OBJS	= $(SRCS:.c=.o)
 OBJS   += $(CPLUSPLUSSRCS:.cpp=.o)
-INCLUDE = -I.. -I../stream -I../loader $(LIBAV_INC)
+INCLUDE = -I.. -I../stream -I../loader $(LIBAV_INC) -I../../nut/libnut
 CFLAGS  = $(INCLUDE) $(OPTFLAGS)
 CPLUSPLUSFLAGS  = $(CFLAGS) -D__STDC_LIMIT_MACROS
 CPLUSPLUS = $(CC)
Index: libmpdemux/demuxer.c
===================================================================
--- libmpdemux/demuxer.c	(revision 19814)
+++ libmpdemux/demuxer.c	(working copy)
@@ -66,6 +66,7 @@
 extern demuxer_desc_t demuxer_desc_lavf;
 #endif
 extern demuxer_desc_t demuxer_desc_aac;
+extern demuxer_desc_t demuxer_desc_nut;
 
 demuxer_desc_t* demuxer_list[] = {
   &demuxer_desc_rawaudio,
@@ -120,6 +121,7 @@
   &demuxer_desc_rawdv,
 #endif
   &demuxer_desc_aac,
+  &demuxer_desc_nut,
 #ifdef HAVE_XMMS
   &demuxer_desc_xmms,
 #endif
Index: libmpdemux/demuxer.h
===================================================================
--- libmpdemux/demuxer.h	(revision 19814)
+++ libmpdemux/demuxer.h	(working copy)
@@ -54,11 +54,12 @@
 #define DEMUXER_TYPE_MPC 40
 #define DEMUXER_TYPE_MPEG_PES 41
 #define DEMUXER_TYPE_MPEG_GXF 42
+#define DEMUXER_TYPE_NUT 43
 
 // This should always match the higest demuxer type number.
 // Unless you want to disallow users to force the demuxer to some types
 #define DEMUXER_TYPE_MIN 0
-#define DEMUXER_TYPE_MAX 42
+#define DEMUXER_TYPE_MAX 43
 
 #define DEMUXER_TYPE_DEMUXERS (1<<16)
 // A virtual demuxer type for the network code
