Index: Makefile
===================================================================
RCS file: /cvsroot/mplayer/main/Makefile,v
retrieving revision 1.341
diff -u -r1.341 Makefile
--- Makefile	16 Feb 2006 20:45:24 -0000	1.341
+++ Makefile	4 Mar 2006 15:46:04 -0000
@@ -42,6 +42,7 @@
               sub_cc.c \
               subreader.c \
               vobsub.c \
+              ../nut/libnut/demuxer.c \
 
 SRCS_MENCODER = mencoder.c \
                 mp_msg-mencoder.c \
Index: libmpdemux/Makefile
===================================================================
RCS file: /cvsroot/mplayer/main/libmpdemux/Makefile,v
retrieving revision 1.101
diff -u -r1.101 Makefile
--- libmpdemux/Makefile	6 Feb 2006 16:04:17 -0000	1.101
+++ libmpdemux/Makefile	4 Mar 2006 15:46:06 -0000
@@ -70,6 +70,7 @@
         demux_mpc.c \
         demux_mpg.c \
         demux_nsv.c \
+        demux_nut.c \
         demux_nuv.c \
         demux_ogg.c \
         demux_pva.c \
@@ -158,7 +159,7 @@
 
 OBJS	= $(SRCS:.c=.o)
 OBJS   += $(CPLUSPLUSSRCS:.cpp=.o)
-INCLUDE = -I.. -I../loader $(CSS_INC) $(EXTRA_INC) $(LIBAV_INC)
+INCLUDE = -I.. -I../loader $(CSS_INC) $(EXTRA_INC) $(LIBAV_INC) -I../../nut/libnut
 CFLAGS  = $(OPTFLAGS) $(INCLUDE) $(XMMS_CFLAGS) $(CDPARANOIA_INC) $(DVB_INC)
 CPLUSPLUSFLAGS  = $(CFLAGS) $(CPLUSPLUSINCLUDE)
 CPLUSPLUS = $(CC)
Index: libmpdemux/demuxer.c
===================================================================
RCS file: /cvsroot/mplayer/main/libmpdemux/demuxer.c,v
retrieving revision 1.217
diff -u -r1.217 demuxer.c
--- libmpdemux/demuxer.c	17 Feb 2006 01:57:41 -0000	1.217
+++ libmpdemux/demuxer.c	4 Mar 2006 15:46:06 -0000
@@ -66,6 +66,7 @@
 extern demuxer_desc_t demuxer_desc_lavf;
 #endif
 extern demuxer_desc_t demuxer_desc_aac;
+extern demuxer_desc_t demuxer_desc_nut;
 
 demuxer_desc_t* demuxer_list[] = {
   &demuxer_desc_rawaudio,
@@ -125,6 +126,7 @@
   &demuxer_desc_lavf,
 #endif
   &demuxer_desc_aac,
+  &demuxer_desc_nut,
   NULL
 };
 
Index: libmpdemux/demuxer.h
===================================================================
RCS file: /cvsroot/mplayer/main/libmpdemux/demuxer.h,v
retrieving revision 1.91
diff -u -r1.91 demuxer.h
--- libmpdemux/demuxer.h	17 Feb 2006 01:57:41 -0000	1.91
+++ libmpdemux/demuxer.h	4 Mar 2006 15:46:07 -0000
@@ -50,11 +50,12 @@
 #define DEMUXER_TYPE_MPC 40
 #define DEMUXER_TYPE_MPEG_PES 41
 #define DEMUXER_TYPE_MPEG_GXF 42
+#define DEMUXER_TYPE_NUT 43
 
 // This should always match the higest demuxer type number.
 // Unless you want to disallow users to force the demuxer to some types
 #define DEMUXER_TYPE_MIN 0
-#define DEMUXER_TYPE_MAX 42
+#define DEMUXER_TYPE_MAX 43
 
 #define DEMUXER_TYPE_DEMUXERS (1<<16)
 // A virtual demuxer type for the network code
